slasher/detection/attestations/spanner.go

```diff
@@ -61,54 +57,48 @@ func (s *SpanDetector) DetectSlashingForValidator(
 	numSpans := uint64(len(s.spans))
 
 	if sp := s.spans[sourceEpoch%numSpans]; sp != nil {
-		minSpan := sp[validatorIdx][0]
+		minSpan := sp[validatorIdx].MinSpan
 		if minSpan > 0 && minSpan < distance {
+			slashableEpoch := sourceEpoch + uint64(minSpan)
+			span := s.spans[slashableEpoch%numSpans][validatorIdx]
 			return &types.DetectionResult{
 				Kind:           types.SurroundVote,
 				SlashableEpoch: sourceEpoch + uint64(minSpan),
+				SigBytes:       span.SigBytes,
 			}, nil
 		}
 
-		maxSpan := sp[validatorIdx][1]
+		maxSpan := sp[validatorIdx].MaxSpan
 		if maxSpan > distance {
+			slashableEpoch := sourceEpoch + uint64(maxSpan)
+			span := s.spans[slashableEpoch%numSpans][validatorIdx]
 			return &types.DetectionResult{
 				Kind:           types.SurroundVote,
-				SlashableEpoch: sourceEpoch + uint64(maxSpan),
+				SlashableEpoch: slashableEpoch,
+				SigBytes:       span.SigBytes,
 			}, nil
 		}
 	}
 
 	if sp := s.spans[targetEpoch%numSpans]; sp != nil {
-		filterNum := sp[validatorIdx][2]
-		if filterNum == 0 {
+		// Check if the validator has attested for this epoch or not.
+		if !sp[validatorIdx].HasAttested {
 			return nil, nil
 		}
-		filterBytes := make([]byte, 2)
-		binary.LittleEndian.PutUint16(filterBytes, filterNum)
-		attFilter := filter.BloomFilter(filterBytes)
 
-		attDataRoot, err := ssz.HashTreeRoot(attData)
-		if err != nil {
-			return nil, err
-		}
-		found, err := attFilter.Contains(attDataRoot[:])
-		if err != nil {
-			return nil, err
-		}
-		if !found {
-			return &types.DetectionResult{
-				Kind:           types.DoubleVote,
-				SlashableEpoch: targetEpoch,
-			}, nil
-		}
+		return &types.DetectionResult{
```

Comment:
seems strange to return double vote when it is not verified

Reply:
This is just for the detection, it gives lets the detection service to do a hard check, which we always need to do anyways for checking double votes.

Reply:
Using 2 bytes of the signature isn't safe enough to just compare the receiving attestation with it IMO, db checks are really quick with this strategy.

Reply:
thats true